diff --git a/src/detection/battery/battery.h b/src/detection/battery/battery.h
index ddeb7c6e..a9ae4c2a 100644
--- a/src/detection/battery/battery.h
+++ b/src/detection/battery/battery.h
@@ -14,6 +14,7 @@ typedef struct FFBatteryResult
     FFstrbuf status;
     FFstrbuf serial;
     double capacity;
+    uint32_t chargeFull;
     double temperature;
     uint32_t cycleCount;
     int32_t timeRemaining; // in seconds, -1 if unknown
diff --git a/src/detection/battery/battery_linux.c b/src/detection/battery/battery_linux.c
index d750dc61..b3d84f40 100644
--- a/src/detection/battery/battery_linux.c
+++ b/src/detection/battery/battery_linux.c
@@ -118,6 +118,16 @@ static void parseBattery(int dfd, const char* id, FFBatteryOptions* options, FFl
         }
     }
 
+    if (ffReadFileBufferRelative(dfd, "charge_full", &tmpBuffer))
+    {
+        int64_t chargeFull = ffStrbufToSInt(&tmpBuffer, 0);
+
+        // Convert to milliamp hours since that is much more readable
+        chargeFull /= 1000;
+
+        result->chargeFull = chargeFull < 0 || chargeFull > UINT32_MAX ? 0 : (uint32_t) chargeFull;
+    }
+
     ffStrbufInit(&result->serial);
     if (ffReadFileBufferRelative(dfd, "serial_number", &result->serial))
         ffStrbufTrimRightSpace(&result->serial);
diff --git a/src/modules/battery/battery.c b/src/modules/battery/battery.c
index 0da4f12d..ec722f82 100644
--- a/src/modules/battery/battery.c
+++ b/src/modules/battery/battery.c
@@ -110,6 +110,7 @@ static void printBattery(FFBatteryOptions* options, FFBatteryResult* result, uin
             FF_ARG(result->modelName, "model-name"),
             FF_ARG(result->technology, "technology"),
             FF_ARG(capacityNum, "capacity"),
+            FF_ARG(result->chargeFull, "charge_full"),
             FF_ARG(result->status, "status"),
             FF_ARG(tempStr, "temperature"),
             FF_ARG(result->cycleCount, "cycle-count"),
@@ -216,6 +217,7 @@ bool ffGenerateBatteryJsonResult(FFBatteryOptions* options, yyjson_mut_doc* doc,
     {
         yyjson_mut_val* obj = yyjson_mut_arr_add_obj(doc, arr);
         yyjson_mut_obj_add_real(doc, obj, "capacity", battery->capacity);
+        yyjson_mut_obj_add_int(doc, obj, "charge_full", battery->chargeFull);
         yyjson_mut_obj_add_strbuf(doc, obj, "manufacturer", &battery->manufacturer);
         yyjson_mut_obj_add_strbuf(doc, obj, "manufactureDate", &battery->manufactureDate);
         yyjson_mut_obj_add_strbuf(doc, obj, "modelName", &battery->modelName);
@@ -277,6 +279,7 @@ FFModuleBaseInfo ffBatteryModuleInfo = {
         {"Battery model name", "model-name"},
         {"Battery technology", "technology"},
         {"Battery capacity (percentage num)", "capacity"},
+        {"Battery full charge capacity (microamp-hours)", "charge_full"},
         {"Battery status", "status"},
         {"Battery temperature (formatted)", "temperature"},
         {"Battery cycle count", "cycle-count"},
