diff --git a/src/filter/tint0r/tint0r.c b/src/filter/tint0r/tint0r.c
index 3632bb5..0fa934b 100644
--- a/src/filter/tint0r/tint0r.c
+++ b/src/filter/tint0r/tint0r.c
@@ -192,27 +192,27 @@ static void tint_sse41(const uint32_t* inframe, uint32_t* outframe, size_t len,
   for (size_t i = 0; i < len; i++)
   {
     /* Load four pixels at once. */
-    p = _mm_loadu_si128((__m128i*)(inframe + i * 4));
+    p = (__m128) _mm_loadu_si128((__m128i*)(inframe + i * 4));
 
     /* Extract four pixels into separate XMM registers and convert them to float. */
-    p0 = _mm_cvtepi32_ps(_mm_cvtepu8_epi32(p));
-    p1 = _mm_cvtepi32_ps(_mm_cvtepu8_epi32(_mm_srli_si128(p, 4)));
-    p2 = _mm_cvtepi32_ps(_mm_cvtepu8_epi32(_mm_srli_si128(p, 8)));
-    p3 = _mm_cvtepi32_ps(_mm_cvtepu8_epi32(_mm_srli_si128(p, 12)));
+    p0 = _mm_cvtepi32_ps(_mm_cvtepu8_epi32((__m128i) p));
+    p1 = _mm_cvtepi32_ps(_mm_cvtepu8_epi32(_mm_srli_si128((__m128i) p, 4)));
+    p2 = _mm_cvtepi32_ps(_mm_cvtepu8_epi32(_mm_srli_si128((__m128i) p, 8)));
+    p3 = _mm_cvtepi32_ps(_mm_cvtepu8_epi32(_mm_srli_si128((__m128i) p, 12)));
 
     #define tint(v) \
       luma = _mm_dp_ps((v), weights, 0x7F); \
       v = _mm_add_ps(_mm_mul_ps(comp_amount, (v)), \
                      _mm_add_ps(_mm_mul_ps(luma, tmp0), tmp1)); \
-      v = _mm_cvtps_epi32(v)
+      v = (__m128) _mm_cvtps_epi32(v)
 
     tint(p0); tint(p1); tint(p2); tint(p3);
 
     /* Gather the processed pixels */
-    p = _mm_packus_epi16(_mm_packus_epi32(p0, p1),
-                         _mm_packus_epi32(p2, p3));
+    p = (__m128) _mm_packus_epi16(_mm_packus_epi32((__m128i) p0, (__m128i) p1),
+                                  _mm_packus_epi32((__m128i) p2, (__m128i) p3));
 
-    _mm_storeu_si128((__m128i*)(outframe + i * 4), p);
+    _mm_storeu_si128((__m128i*)(outframe + i * 4), (__m128i) p);
   }
 }
 #elif USE_AVX2
